### DISCUSSION TITLE
Extension - Public Web Page / Components

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#10934] [Feature Requests]
Hi - is there a way to register a public root webpage which is presented as default in “/“.

The current behavior is that any root requests are redirected to “/login”. And it’s also not possible to create such a module as a module-id is required… which is then equivalent to the path.

What would be amazing is that directus checks whether a specific module with specific name,…. exists… if it exists it presents it as public root page… if not it continuous to default / redirect to /login

thanks for any thoughts in advance.

PS - an extension of this would be to allow to modify the reset password, accept invite,… pages via similar mechanisms… currently I cannot modify those…

**Comment by rijkvanzanten** [comment]
There's currently no way to create custom public-pages

**Comment by sebj54** [comment]
Actually you can, but it is a bit "tricky". **You cannot replace existing pages though.**

You can achieve it by creating a custom endpoint with a JS file (for instance `extensions/endpoints/test/index.js`:

```js
module.exports = (router) => {
    router.get('/', async (req, res) => {
        res.type('html')
        res.end('<h1>Hello world!</h1>')
    })
}
```

Then if yout hit `http://localhost:8055/test`, you'll have a public HTML page.

**Reply by joselcvarela** [reply]
That's why this tool is so amazing ❤️
You can also use custom API Hooks for that. You can even replace the `/admin` endpoint and use a custom login page using SDK for example 😁

**Reply by rijkvanzanten** [reply]
I stand corrected!

**Reply by sebj54** [reply]
So you can replace existing pages? 😱 I may have failed when I tested it 😅 

> That's why this tool is so amazing ❤️

Totally agree!

**Comment by ernst86913** [comment]
Sounds great.

How am I able to replace / and /admin via custom endpoint?

Thanks a lot for your help in advance 

LG,
Roman

Von meinem iPhone gesendet

**Reply by joselcvarela** [reply]
You can use `ROOT_REDIRECT` with the solution provided by sebj54 to achieve this:
https://docs.directus.io/configuration/config-options/#:~:text=100kb-,ROOT_REDIRECT,-Where%20to%20redirect
https://github.com/directus/directus/discussions/10934#discussioncomment-1999117

In that case you should `ROOT_REDIRECT=./test`

It is possible to not serve the app if that's what you are looking for by setting `SERVE_APP=false`

If you just want to override `/admin` that's a more complex scenario because the App relies on this and you may incur into issues. But with `init('middlewares.before')` you may override the `app.get('^/admin$' (req, res)=> res.redirect(env.ROOT_REDIRECT))`

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
